src/utilities/compute_spatially_variant_penalty_strength.cxx

```diff
@@ -0,0 +1,200 @@
+      std::cerr<<"\nUsage: compute_spatially_variant_penalty_strength template_proj_data\n";
+      exit(EXIT_FAILURE);
+    }
+END_NAMESPACE_STIR
+
+USING_NAMESPACE_STIR
+class KappaComputation: public ParsingObject
+{
+public:
+    KappaComputation();
+    void set_defaults();
+    Succeeded run();
+    typedef DiscretisedDensity<3,float> target_type;
+
+protected:
+    shared_ptr<GeneralisedObjectiveFunction<target_type> >  objective_function_sptr;
+
+    // This does something to do with saving, ill come back to it
+    shared_ptr<OutputFileFormat<DiscretisedDensity<3,float> > > output_file_format_sptr;
+
+    Succeeded compute_kappa_at_current_image_estimate();
+    Succeeded compute_kappa_with_approximate();
+
+private:
+    std::string input_image_filename;
+    std::string template_image_filename;
+    std::string kappa_filename;
+    void initialise_keymap();
+    bool post_processing();
+};
+
+KappaComputation::KappaComputation()
+{
+  set_defaults();
+}
+
+
+void
+KappaComputation::set_defaults()
+{
+  objective_function_sptr.reset(new PoissonLogLikelihoodWithLinearModelForMeanAndProjData<target_type>);
+}
+
+void
+KappaComputation::initialise_keymap()
+{
+  parser.add_start_key("Kappa Computation Parameters");
+  parser.add_key("kappa filename", &kappa_filename);
+  parser.add_key("input image", &input_image_filename);
+  parser.add_key("template image", &template_image_filename);
+  parser.add_parsing_key("objective function type", &objective_function_sptr);
+  parser.add_stop_key("End");
+}
+
+bool
+KappaComputation::post_processing()
+{
+  if (is_null_ptr(this->objective_function_sptr))
+  {
+    error("objective_function_sptr is null");
+    return true;
+  }
+
+  if (input_image_filename.empty() && template_image_filename.empty())
+  {
+    error("Requires either input_image_filename or template_image_filename");
+    return true;
+  }
+  return false;
+}
+
+Succeeded
+KappaComputation::run()
+{
+  if (!input_image_filename.empty())
+  {
+    compute_kappa_at_current_image_estimate();
+  }
+  else if (!template_image_filename.empty())
+  {
+    compute_kappa_with_approximate();
+  }
+  return Succeeded::yes;
+}
+
+
+Succeeded
+KappaComputation::compute_kappa_at_current_image_estimate()
+{
+  info("Computing the spatially variant penalty strength at the current image estimate.");
+  shared_ptr<DiscretisedDensity<3,float>>
+          current_image_estimate(read_from_file<DiscretisedDensity<3,float>>(input_image_filename));
+  current_image_estimate->get_data_ptr();
+
+  shared_ptr<DiscretisedDensity<3, float>> output_image;
+  output_image = static_cast<const shared_ptr<DiscretisedDensity<3, float>>>(current_image_estimate->get_empty_copy());
+  output_image->fill(0.);
+
+  shared_ptr<DiscretisedDensity<3, float>> ones_image;
+  ones_image = static_cast<const shared_ptr<DiscretisedDensity<3, float>>>(current_image_estimate->get_empty_copy());
+  ones_image->fill(1.);
+
+  objective_function_sptr->set_up(current_image_estimate);
```

Comment:
This currently needs setup because of a check on image characteristics in the sub functionality of `accumulate_Hessian_times_input` -> https://github.com/robbietuk/STIR/blob/5aa14155e728ced619e3ce1f1a7b8d4b9eb45422/src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndProjData.cxx#L818-L819